src/main/java/it/gov/pagopa/payment/service/payment/idpaycode/IdpayCodePreAuthServiceImpl.java

```diff
@@ -57,8 +59,7 @@ public RelateUserResponse relateUser(String trxId, String fiscalCode) {
         String userId = retrieveUserId(fiscalCode);
 
         TransactionInProgress trx = transactionInProgressRepository.findById(trxId)
-                .orElseThrow(() -> new ClientExceptionWithBody(
-                        HttpStatus.NOT_FOUND,
+                .orElseThrow(() -> new TransactionNotFoundOrExpiredException(
```

Comment:
some Exception, like this, could have a default code that is supposed to be always the same: we could thus expose a constructor without CODE, using such default code in the constructor